src/System/Relay/middleware/cdnCacheStatusMiddleware.ts

```diff
@@ -0,0 +1,24 @@
+import { updateContext } from "Server/middleware/bootstrapSharifyAndContextLocalsMiddleware"
+
+export const cdnCacheStatusCacheKey = (queryName: string, variables: any) => {
+  return `cache-status:${queryName}:${JSON.stringify(variables)}`
+}
+
+export const cdnCacheStatusMiddleware = () => {
+  return next => async req => {
+    const res = await next(req)
+
+    const cacheStatus = res._res.headers.get("cf-cache-status") === "HIT"
+
+    const isRouteQuery = /^.*Routes_/.test(req.operation.name as string)
+
+    if (isRouteQuery) {
+      updateContext(
```

Comment:
Our wrapper of `asyncLocalStorage` already works isomorphically! https://github.com/artsy/force/blob/fbc48823c4add465ee20f82ebfa352948d300c7b/src/Server/asyncLocalWrapper/index.ts#L3

It's a blunt tool to be sure, but feels reasonable as these values need to be propagated from Relay middleware, into analytics hooks - otherwise seems like a pain/refactor to try to thread the needle otherwise.